service/gsheet_bot.py

```diff
@@ -0,0 +1,167 @@
+import gspread
+from google.oauth2.service_account import Credentials
+import csv
+import os
+from config import Config
+from datetime import datetime
+from discord.utils import get
+
+config = Config()
+
+# Google Sheet Configurations
+CREDENTIALS_FILE = 'config/services_account.json'
+
+# System Configurations
+GBOT_MINIMUM_VOLUME_THRESHOLD = int(os.getenv('GBOT_MINIMUM_VOLUME_THRESHOLD', 30000))
+
+# Output Configurations
+month_year = datetime.now().strftime("%b%Y")
+OUTPUT_FOLDER_NAME = '../output'
+OUTPUT_VOLUME_FILENAME = f'{OUTPUT_FOLDER_NAME}/volume_{month_year}.json'
+OUTPUT_PASS_VOLUME_CSV_FILENAME = f'{OUTPUT_FOLDER_NAME}/pass_volume_users_{month_year}.csv'
+OUTPUT_FAIL_VOLUME_CSV_FILENAME = f'{OUTPUT_FOLDER_NAME}/fail_volume_users_{month_year}.csv'
+
+
+async def batch_update_trade_volume_table(dbcon, guild):
+    role_name = 'FREE_VIP'
+    worksheet = get_worksheet(config.GBOT_GSHEET_ID)
+    all_users_volume = get_all_user_volume(worksheet)
+    # write_csv_for_volume_users(all_users_volume)
+
+    if not all_users_volume:
+        print("No data found. Skipping update.")
+        return False
+
+    for user_dict in all_users_volume:
+        if not user_dict:
+            print("Empty row found. Skipping row.")
+            continue
+
+        for uuid, user_data in user_dict.items():
+            if not uuid or not user_data or not str(uuid).strip():
+                print("Empty UUID or user data found. Skipping entry.")
+                continue
+
+            discord_id = user_data['discord_id']
+            if not discord_id or not str(discord_id).strip():
+                print("Empty or invalid Discord ID found. Skipping entry.")
+                continue
+
+            volume = user_data['volume']
+            remark = user_data['remark']
+            status = user_data['status']
+
+            if dbcon.fetch_user_trade_volume_by_discord_id(discord_id):
+                print('Update Discord ID: ', discord_id)
+                dbcon.update_trade_volume_table_by_discord_id(discord_id, volume, remark, status)
```

Comment:
i think we can try consolidate them into one big string and update with 1 db transaction
```
INSERT INTO tbl_name
    (a,b,c)
VALUES
    (1,2,3),
    (4,5,6),
    (7,8,9);